src/model/keychain.js

```diff
@@ -136,3 +143,86 @@ export async function hasKey(key) {
   }
   return null;
 }
+
+export async function restoreBackupIntoKeychain(backedUpData) {
+  // Access control config per each type of key
+  const publicAccessControlOptions = {
+    accessible: ACCESSIBLE.ALWAYS_THIS_DEVICE_ONLY,
+  };
+
+  let privateAccessControlOptions = {};
+  const canAuthenticate = await canImplyAuthentication({
+    authenticationType: AUTHENTICATION_TYPE.DEVICE_PASSCODE_OR_BIOMETRICS,
+  });
+
+  let isSimulator = false;
+
+  if (canAuthenticate) {
+    isSimulator = __DEV__ && (await DeviceInfo.isEmulator());
+  }
+  if (canAuthenticate && !isSimulator) {
```

Comment:
Lines 153 - 168: this exists in multiple places in model/wallet.js as well. Perhaps we should separate it out into its own function for determining the privateAccessControlOptions